notebooks/214-vehicle-detection-and-recognition/214-vehicle-detection-and-recognition.ipynb

```diff
@@ -0,0 +1,545 @@
+    "ie_core = IECore()\n",
+    "\n",
+    "\n",
+    "def model_init(model: str) -> Tuple:\n",
+    "    \"\"\"\n",
+    "    Read the network and weights from file, load the\n",
+    "    model on the CPU and get input and output names of nodes\n",
+    "\n",
+    "    :param: model: model architecture path *.xml\n",
+    "    :retuns:\n",
+    "            input_key: Input node network\n",
+    "            output_key: Output node network\n",
+    "            exec_net: Encoder model network\n",
+    "            net: Model network\n",
+    "    \"\"\"\n",
+    "\n",
+    "    # Read the network and corresponding weights from file\n",
+    "    net = ie_core.read_network(model)\n",
+    "    # load the model on the CPU (you can use GPU or MYRIAD as well)\n",
+    "    exec_net = ie_core.load_network(net, \"CPU\")\n",
+    "    # Get input and output names of nodes\n",
+    "    input_keys = list(exec_net.input_info)\n",
+    "    output_keys = list(exec_net.outputs.keys())\n",
+    "    return input_keys, output_keys, exec_net, net"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "95c97641",
+   "metadata": {},
+   "source": [
+    "### Get attributes from model\n",
+    "\n",
+    "We use `XXX.input_info.tensor_desc.dims` to get data shapes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2d864515",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# de -> detection\n",
+    "# re -> recognition\n",
+    "# Detection model initialization\n",
+    "input_key_de, output_keys_de, exec_net_de, net_de = model_init(detection_model_path)\n",
+    "# Recognition model initialization\n",
+    "input_key_re, output_keys_re, exec_net_re, net_re = model_init(recognition_model_path)\n",
+    "\n",
+    "# Get input size - Detection\n",
+    "height_de, width_de = net_de.input_info[input_key_de[0]].tensor_desc.dims[2:]\n",
+    "# Get input size - Recognition\n",
+    "height_re, width_re = net_re.input_info[input_key_re[0]].tensor_desc.dims[2:]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bb050ef3",
+   "metadata": {},
+   "source": [
+    "### Helper function\n",
+    "\n",
+    "1. `plt_show` function is used to show image \n",
+    "2. `softmax` function is a generalization of the logistic function to multiple dimensions to normalize the output of a network to a probability distribution over predicted output classes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "474826d8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plt_show(raw_image):\n",
+    "    \"\"\"\n",
+    "    Use matplot to show image inline\n",
+    "    raw_image: input image\n",
+    "\n",
+    "    :param: raw_image:image array\n",
+    "    \"\"\"\n",
+    "    plt.figure(figsize=(10, 6))\n",
+    "    plt.axis(\"off\")\n",
+    "    plt.imshow(raw_image)\n",
+    "\n",
+    "\n",
+    "def softmax(x: np.ndarray) -> np.ndarray:\n",
```

Comment:
You don't need this function. As stated here: https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/intel/vehicle-attributes-recognition-barrier-0039 the model has its own softmax inside.